compass/ocean/tests/isomip_plus/initial_state.py

```diff
@@ -125,12 +129,20 @@ def _compute_initial_condition(self):
 
         ds['landIceFraction'] = xr.where(mask, ds.landIceFraction, 0.)
 
-        ref_density = constants['SHR_CONST_RHOSW']
-        landIcePressure, landIceDraft = compute_land_ice_pressure_and_draft(
-            ssh=ds.ssh, modify_mask=ds.ssh < 0., ref_density=ref_density)
+        land_ice_draft = ds.ssh
```

Comment:
The reason I don't do this for the thin film case in Polaris is because the ocean won't want to have the SSH at the flotation height that BISICLES wants to give it.  This is at least partly because BISICLES used a different reference density for the ocean (1028 instead of 1026) and partly because it has been my experience that computing the ice draft (SSH) from the land-ice pressure and the reference ocean density tends to produce the most reliable starting point for the SSH adjustment step.

Reply:
I'm not sure if this will help with the state validation issues during SSH adjustment for the thin-film case but I think it might.

Reply:
I discovered that I was missing a check in my Polaris computation of land_ice_draft:
```python
            land_ice_draft = np.maximum(ds_topo.bedrockTopography,
                                        land_ice_draft)
```
We obviously don't what to allow the ice draft to go deeper than the bedrock in the thin film region.

Reply:
If that's the case, then wouldn't we want to assign the ssh to the recomputed land ice draft for both thin film and non-thin film cases?

Reply:
We could certainly do that.  It would dovetail better with the coupling strategy.

The reason I had been working with SSH/draft as the known quantity and land-ice pressure as the one to adjust was that this is more how we operate with static cavities with realistic topography (e.g. from  BedMachine).  Without wetting and drying, you can end up with too thin a water column if you are using land-ice pressure to determine SSH.  But we could either adjust the grounding line if this happens.  We can't adjust the minimum column thickness unless we also change the land-ice pressure because otherwise the pressure will just slam the column back down.

Reply:
I modify the bottom depth to maintain min water column thickness here: https://github.com/MPAS-Dev/compass/blob/b977252d360f52d9e8dc7c2af2d7c5efefe143ec/compass/ocean/tests/isomip_plus/initial_state.py#L156-L167 But I only do this for the initial state so it wouldn't guarantee for a time-varying pressure that the water column thickness would stay above the threshold without wetting and drying.

Reply:
Right.  In my experience, the only to have things behave robustly without wetting-and-drying has been to treat the water-column thickness as the thing you know and the land-ice pressure as the thing you can adjust (in the poorly named "ssh adjustment" step).  Even if you thicken the water column in the initial condition, if the land-ice pressure is too high, it will just slam right back down and try to evacuate the water column.